controller/volume_controller.go

```diff
@@ -2302,12 +2302,24 @@ func (vc *VolumeController) updateRequestedBackupForVolumeRestore(v *longhorn.Vo
 	if v.Spec.FromBackup == "" {
 		return nil
 	}
-	if !v.Status.RestoreRequired && !v.Status.IsStandby {
+
+	// If the volume is not restoring, skip setting e.Spec.RequestedBackupRestore
+	if !v.Status.RestoreRequired {
 		return nil
 	}
 
-	if v.Status.LastBackup != "" && v.Status.LastBackup != e.Spec.RequestedBackupRestore {
+	// For DR volume we set RequestedBackupRestore to the LastBackup
+	if v.Status.IsStandby && v.Status.LastBackup != "" && v.Status.LastBackup != e.Spec.RequestedBackupRestore {
 		e.Spec.RequestedBackupRestore = v.Status.LastBackup
+		return nil
+	}
+
+	// For non-DR restoring volume we set RequestedBackupRestore to the backup in v.Spec.FromBackup
+	if v.Spec.FromBackup != "" {
```

Comment:
NIT: This field won't be empty. See line 2302 above.

Reply:
Thank you. Fixed